Makefile

```diff
@@ -1,13 +1,222 @@
+BUMP_XARGS := $(call clean_opt,$(BUMP_XARGS))
+ifeq ($(filter dry, $(MAKECMDGOALS)), dry)
+  BUMP_XARGS := $(BUMP_XARGS) --dry-run
+endif
+BUMP_CFG  ?= .bumpversion.cfg				## Bump version configuration (default recommended)
+BUMP_CFG  := $(call clean_opt,$(BUMP_CFG))
+BUMP_TOOL := bump2version
+BUMP_CMD  := $(BUMP_TOOL) --config-file "$(BUMP_CFG)"
+
+# guess the applicable semantic level update if provided via major|minor|patch targets
+# perform validation to avoid many combination provided simultaneously
+override BUMP_VERSION_INPUT := 1
+override BUMP_VERSION_LEVEL :=
+ifeq ($(filter major, $(MAKECMDGOALS)), major)
+  ifeq ($(BUMP_VERSION_LEVEL),)
+    override BUMP_VERSION_LEVEL := major
+    override BUMP_VERSION_INPUT := 0
+  else
+    $(error "Cannot combine 'major|minor|patch' simultaneously!")
+  endif
+endif
+ifeq ($(filter minor, $(MAKECMDGOALS)), minor)
+  ifeq ($(BUMP_VERSION_LEVEL),)
+    override BUMP_VERSION_LEVEL := minor
+    override BUMP_VERSION_INPUT := 0
+  else
+    $(error "Cannot combine 'major|minor|patch' simultaneously!")
+  endif
+endif
+ifeq ($(filter patch, $(MAKECMDGOALS)), patch)
+  ifeq ($(BUMP_VERSION_LEVEL),)
+    override BUMP_VERSION_LEVEL := patch
+    override BUMP_VERSION_INPUT := 0
+  else
+    $(error "Cannot combine 'major|minor|patch' simultaneously!")
+  endif
+endif
+ifneq ($(VERSION),)
+  ifneq ($(BUMP_VERSION_LEVEL),)
+    $(error "Cannot combine 'major|minor|patch' simultaneously with explicit VERSION input!")
+  endif
+endif
+# when none was provided, use patch to directly apply the explicit 'VERSION' specified as input
+ifeq ($(BUMP_VERSION_LEVEL),)
+  override BUMP_VERSION_LEVEL := --new-version "$(VERSION)" patch
+endif
+
+.PHONY: dry
+dry: $(BUMP_CFG)
+	@-$(ECHO) > /dev/null
+
+.PHONY: patch
+patch: $(BUMP_CFG)
+	@-$(ECHO) > /dev/null
+
+.PHONY: minor
+minor: $(BUMP_CFG)
+	@-$(ECHO) > /dev/null
+
+.PHONY: major
+major: $(BUMP_CFG)
+	@-$(ECHO) > /dev/null
+
+
+# run bumpversion with replacement of the release time value within its config for auto-update on future revisions
+.PHONY: bump
+bump: bump-check bump-install  ## Bump version using specified <VERSION> (call: make VERSION=<MAJOR.MINOR.PATCH> bump)
+	@-$(MSG_I) "Updating package version..."
+	@[ $(BUMP_VERSION_INPUT) -eq 0 ] || [ "${VERSION}" ] || ( \
+		$(MSG_E) "Argument 'VERSION' is not specified to bump version"; exit 1 \
+	)
+	@$(SHELL) -c ' \
+		PRE_RELEASE_TIME=$$(head -n 1 RELEASE.txt | cut -d " " -f 2) && \
+		$(BUMP_CMD) $(BUMP_XARGS) $(BUMP_VERSION_LEVEL) && \
+		POST_RELEASE_TIME=$$(head -n 1 RELEASE.txt | cut -d " " -f 2) && \
+		echo "Replace $${PRE_RELEASE_TIME} → $${POST_RELEASE_TIME}" && \
+		$(_SED) -i "s/$${PRE_RELEASE_TIME}/$${POST_RELEASE_TIME}/g" $(BUMP_CFG) && \
+		git add $(BUMP_CFG) && \
+		git commit --amend --no-edit \
+	'
+
+.PHONY: bump-install
+bump-install:   ## Installs bump2version if not detected in the environment
+	@-$(SHELL) -c 'which -s "$(BUMP_TOOL)" || pip install $(BUMP_TOOL)'
```

Comment:
The -s flag for which is not defined in some shells. If you want to silence stdout and stderr:

```
which "$(BUMP_TOOL)" &> /dev/null
```

or if we want to avoid using `which` entirely:

```
python -m "$(BUMP_TOOL) -h &> /dev/null || python -m pip install $(BUMP_TOOL)